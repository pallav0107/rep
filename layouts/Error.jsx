import React from 'react';
import styled from 'styled-components';
import { Flex } from 'grid-styled';
import { Container, Backdrop } from './common';
import { RouterButton } from '../components/RouterButton';

export const ErrorContainer = styled(Flex)`
  border: 1px ${({ theme }) => theme.color.border} solid;
  border-radius: 3px;
  background-color: #fff;
  width: 100%;
  height: 100%;
  padding-top: 30px;
  padding-bottom: 40px;
  flex-direction: column;
  align-items: center;
`;

const Error = () => (
  <Container column flex="1 0 auto">
    <Backdrop align="center" justify="flex-start" p={40} column>
      <ErrorContainer>
        <h1>404 - Page Not Found</h1>
        <RouterButton to="/" label="Home" />
      </ErrorContainer>
    </Backdrop>
  </Container>
);

export default Error;
