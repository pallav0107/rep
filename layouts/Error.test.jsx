import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Error, { ErrorContainer } from './Error';
import { withTheme } from '../../utils/jest/TestUtils';
import "../setupTests"

describe('Error page', () => {
  it('renders the Error page', () => {
    const component = withTheme(<Error />);
    const wrapper = shallow(component).dive();

    expect(wrapper).toMatchSnapshot();
  });

  describe('testing styled-components with theming', () => {
    it('should render ErrorIcon', () => {
      const component = withTheme(<ErrorContainer />);
      const wrapper = renderer.create(component).toJSON();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
