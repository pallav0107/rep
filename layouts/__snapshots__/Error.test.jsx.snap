// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page renders the Error page 1`] = `
<Styled(Flex)
  column={true}
  flex="1 0 auto"
>
  <Styled(Flex)
    align="center"
    column={true}
    justify="flex-start"
    p={40}
  >
    <Styled(Flex)>
      <h1>
        404 - Page Not Found
      </h1>
      <withRouter(RouterButton)
        label="Home"
        to="/"
      />
    </Styled(Flex)>
  </Styled(Flex)>
</Styled(Flex)>
`;

exports[`Error page testing styled-components with theming should render ErrorIcon 1`] = `
.c2 {
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  border: 1px #d7d2cb solid;
  border-radius: 3px;
  background-color: #fff;
  width: 100%;
  height: 100%;
  padding-top: 30px;
  padding-bottom: 40px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div
  className="c0 c1 c2"
  is={null}
/>
`;
